src/org/org.ts

```diff
@@ -655,7 +655,7 @@ export class Org extends AsyncOptionalCreatable<Org.Options> {
     const username = this.getUsername();
     if (username) {
       const organization = await this.retrieveOrganizationInformation();
-      const isScratch = organization.IsSandbox && organization.TrialExpirationDate;
+      const isScratch = organization.IsSandbox && Boolean(organization.TrialExpirationDate);
```

Comment:
surprising that it's not `!organization.IsSandbox`. Seems like `.IsSandbox` should not be `true` when it's a scratch org... so continues the difficulties of determining scratch orgs

Reply:
It's also weird that the `&&` in the middle doesn't turn the whole expression into a boolean, thanks JavaScript.
![image](https://user-images.githubusercontent.com/1084688/176275659-57601054-e53b-4b64-8b8f-1634f2ba45d9.png)

Also a place where TypeScript can help 😄 
![image](https://user-images.githubusercontent.com/1084688/176276026-8cc338c7-8040-427b-bea2-f3838eb902d6.png)

Reply:
I guess `IsSandbox ? it's a sandbox : it's a prod org` no options for scratch orgs :)

Reply:
yep

Reply:
Mike fixed some typing elsewhere to prevent this from happening.  I merged those in.